okhttp/src/jvmMain/kotlin/okhttp3/internal/http/CallServerInterceptor.kt

```diff
@@ -103,8 +103,10 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
           .receivedResponseAtMillis(System.currentTimeMillis())
           .build()
       var code = response.code
-      if (code == 100) {
-        // Server sent a 100-continue even though we did not request one. Try again to read the
+      while (code in 100..199) {
+        exchange.eventListener.responseHeadersEnd(exchange.call, response)
+
+        // Server sent a 1xx so report and continue. Try again to read the
```

Comment:
Seems like I shouldn't ignore a "Switching Protocols" here. 🤔

Reply:
I think it's a future-proof default. What does curl do with HTTP 105??

Reply:
I assume this is broken for 101?